src/frequenz/client/base/channel.py

```diff
@@ -51,6 +51,16 @@ class ChannelOptions:
     ssl: SslOptions = SslOptions()
     """SSL options for the channel."""
 
+    keep_alive_time_ms: int | None = None
+    """The time in milliseconds between HTTP2 keep-alive pings.
+
+    If None, keep-alive is disabled."""
+
+    keep_alive_timeout_ms: int = 20_000
+    """The time in milliseconds to wait for a HTTP2 keep-alive response.
+
+    This is only used if `keep_alive_time_ms` is not None."""
```

Comment:
I would use `timedelta` here, as we try to keep high-level types when facing the users (developers).

Reply:
I would mimic how `SslOptions` works. So, adding a separate class for `KeepAliveOptions`, with a `enabled` config option like with to be able to disabled the whole thing and still provide different defaults in case the end user wants to enable it query params, so also allowing overrides via query params, and allow configuring all options instead of leaving some hardcoded (we can provide the current values as defaults).

Reply:
I've added `KeepAliveOptions`.  How should the query parameters specify interval?

Reply:
Good question. I guess a float with seconds?